index.d.ts

```diff
@@ -1,13 +1,19 @@
 /* tslint:disable:interface-name max-classes-per-file no-empty-interface */
 
 import * as React from 'react'
-import Box, { extractStyles as boxExtractStyles } from 'ui-box'
-import { BoxProps } from 'ui-box/dist/types/box-types'
+import { extractStyles as boxExtractStyles, BoxProps, BoxOwnProps } from 'ui-box'
 import { StyleAttribute, CSSProperties } from 'glamor'
 import { DownshiftProps } from 'downshift'
 import { TransitionProps, TransitionStatus } from 'react-transition-group/Transition'
 
-export { configureSafeHref } from 'ui-box'
+export { configureSafeHref, BoxProps } from 'ui-box'
+
+/**
+ * Convenience method for defining your own components that extend Box and pass-through props
+ */
+export type BoxComponent<P = {}, D extends React.ElementType = React.ElementType> = <
```

Comment:
Clever! Looks legit -- take component-specific props, take an `is` prop, and do some exclusion on those `BoxProps` and the component-specific props.

Reply:
The big win here is that this type respects consumers passing in other `is` values and it will give them all the props along with that!